### DISCUSSION TITLE
Allow qmd sections / subsections to be enclosed in <details> tags in HTML documents.

### DISCUSSION THREAD

**Discussion by aaronsgithub** [quarto-dev/quarto-cli#5523] [Q&A] [answered]
We can use the html `<details>` tag feature to fold the visibility of code blocks and choose their default visibility:

    ```{r}
    #| code-fold: true
    #| code-summary: "Show the code"
    ```
It would be a nice feature if you could specify a section to be rendered as a `<details>` :
```
## Section 2

### Section 2 contents

#### Interlude: Background material  <- syntax to specify as <details>
This content is foldable

### More section 2 contents

## Section 3
```
This could be useful for tangential or interlude sections of documents which are not part of the main contents of the document, but may be folded away and accessed via a toggle.

By doing this at a section level, it would mean that any markdown and code blocks within that section would be folded away as one logical group.

Is there any way of achieving this with the current version or is this something that would have to be implemented if desired?

**Comment by mcanouil** [comment, marked as answer]
I don't think that's a good idea.
If the section is not really part of the main document, why not put it in an appendix and make a reference to it?
Or use conditional content to hide/show section depending on condition or profile.

This being said, there is nothing to add in Quarto, everything is already there, _i.e._, Bootstrap (see <https://getbootstrap.com/docs/5.2/components/collapse/>).

<details><summary>See Quarto document code.</summary>

````qmd
---
title: "Quarto Playground"
format: html
---

This is a playground for Quarto.


## Section 1

lore ipsum dolor sit amet 1

## Section 2 <a class="btn btn-primary" data-bs-toggle="collapse" href="#multiCollapseExample1" role="button" aria-expanded="false" aria-controls="multiCollapseExample1">Show</a>

::: {#multiCollapseExample1 .collapse .multi-collapse}

### Subsection 1

lore ipsum dolor sit amet 2

:::


## Section 3

lore ipsum dolor sit amet 3

````

</details>

https://github.com/quarto-dev/quarto-cli/assets/8896044/adea52fd-2de1-4a44-969f-b879f65a9b91

**Reply by mcanouil** [reply]
I am going to convert this as a Q&A rather than a Feature Request.

**Comment by aaronsgithub** [comment]
The desire to have it as an optional section within the main content rather than an appendix was only motivated as a UX design choice for that particular section.

But what you've demonstrated solves the problem, so thank you!
